odoo/addons/base/models/ir_profile.py

```diff
@@ -108,3 +118,25 @@ def set_profiling(self, profile=None, collectors=None, params=None):
             'collectors': request.session.profile_collectors,
             'params': request.session.profile_params,
         }
+
+
+class EnableProfilingWizard(models.TransientModel):
+    _name = 'base.enable.profiling.wizard'
+    _description = "Enable profiling for some time"
+
+    duration = fields.Selection([
+        ('minutes_5', "5 Minutes"),
+        ('hours_1', "1 Hour"),
+        ('days_1', "1 Day"),
+        ('months_1', "1 Month")
+    ], string="Enable profiling for")
+    expiration = fields.Datetime("Enable profiling until", compute='_compute_expiration', store=True, readonly=False)
+
+    @api.depends('duration')
+    def _compute_expiration(self):
+        for record in self:
```

Comment:
Given this is a single-record wizard, the loop can probably be done without?

Reply:
I think it is a good practice to always make code work in multi record when it is possible. But i can remove it if you don't like that.

Reply:
Please keep the for loop.  The ORM will always try to invoke the compute method on a batch of records.  Most of the time, it will be a single record in this case.  But it is not guaranteed.

Reply:
> Please keep the for loop. The ORM will always try to invoke the compute method on a batch of records. Most of the time, it will be a single record in this case. But it is not guaranteed.

I don't really see a situation where this would be batch-computed, possibly aside from a test batch-creating a bunch of those but... why?